### ISSUE TITLE
Passing a different referrer value creates a new visit

### ISSUE THREAD

**Issue by d4rken** [piwik/piwik#7645]
Is it possible that if i change the referrer during the queries of one visit, the new referrer gets a new visit entry?

Case:
I have a visit and during this visit i send two download events to be tracked. If both download events get a distinct referrer parameter, i have two visits in my visitor log.
The main visit with all entries (and my first download and it's referrer)
and a secondary visit with only one entry that had the different referrer.

TL;DR is there a "one referrer per visit rule" ?

**Comment by dotsbb** [comment]
@mattab Could You look at this?

**Comment by mattab** [comment]
Here is how it should work, as documented in this FAQ: https://piwik.org/faq/how-to/faq_19616/

 * By default Piwik will create a new visit whenever a visitor views your website using a different campaign information from the visitor’s ongoing visit. You can disable this behavior by setting create_new_visit_when_campaign_changes=0 under [Tracker] section in your config/config.ini.php file.

 * By default, Piwik won’t create a new visit when a visitor views your website using a different website referrer information. You may configure Piwik to force it to create a new visit whenever a visitor uses a new website referrer. 

if you find it does not work in this way as you'd expect, please reopen the issue
